### PR TITLE
[MINOR] build: Bump the version to 0.7.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

Bump the version to 0.7.0-incubating-SNAPSHOT and 0.7.0.dev for Java and Python build.

For Python, I checked several incubating projects, they don't have "incubating" in version.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing CIs.

### CODE REVIEW

**Comment by jerryshao** [comment]
Please hold the merge, I still need to fix something.

File: clients/client-python/gravitino/exceptions/base.py [hunk 1]
```diff
@@ -101,4 +101,3 @@ class UnauthorizedException(GravitinoRuntimeException):
 
 class BadRequestException(GravitinoRuntimeException):
     """An exception thrown when the request is invalid."""
-
```
Comment (yuqi1129) [inline clients/client-python/gravitino/exceptions/base.py:104]: Is this change expected?

Comment (jerryshao) [reply clients/client-python/gravitino/exceptions/base.py:104]: Yes, it's a formatting issue.

Comment (yuqi1129) [reply clients/client-python/gravitino/exceptions/base.py:104]: OK

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**